prov/efa/src/rdm/efa_rdm_peer.c

```diff
@@ -98,10 +98,14 @@ void efa_rdm_peer_destruct(struct efa_rdm_peer *peer, struct efa_rdm_ep *ep)
 		efa_rdm_txe_release(txe);
 	}
 
-	dlist_foreach_container_safe(&peer->rxe_list,
-				     struct efa_rdm_ope,
-				     rxe, peer_entry, tmp) {
-		efa_rdm_rxe_release(rxe);
+	if (!dlist_empty(&peer->rxe_list)) {
```

Comment:
should we just lock the whole peer destruct?

Reply:
Updated